Omr installation docs (#39593)
openshift/openshift-docs: modules/mirror-registry-localhost.adoc

[NOTE]
====
Installing Red Hat Quay using the `mirror-registry` CLI tool makes several changes to your machine. After installation, a `/etc/quay-install` directory is created, which has installation files, local storage, and the configuration bundle. Trusted SSH keys are generated in case the deployment target is the local host, and systemd files on the host machine are set up to ensure that container runtimes are persistent. Additionally, an initial user named `init` is created with an auto-generated password. All access credentials are printed at the end of the install routine.
====

.Procedure

. Download the `mirror-registry.tar.gz` package for the latest version of mirror registry found on the link:https://console.redhat.com/openshift/downloads[*Downloads*] page of {product-title}.

. Install the mirror registry on your local host with your current user account by using the `mirror-registry` tool:
+
[source,terminal]
----
$ sudo ./mirror-registry install \
  --quayHostname <host_example_com> \
  --quayRoot <example_directory_name>

> ----

Reviewer: Lets add notes below this that point back to https://github.com/openshift/openshift-docs/pull/39593/files#diff-69cfe3d2263e019a7a1c1e939ecd40e84d804eff718d709a4da103d244f1792f and show users what options they can set; to over additional configuration during install.

. Use the user name and password generated during installation to log into the registry by running the following command:
+
[source,terminal]
----
$ podman login -u init \
  -p <password> \

>   --tls-verify=false <1>

Reviewer 2: ```suggestion
  --tls-verify=false <1>
  <host_example_com>:8443
```

This is missing the registry to login to.

// Module included in the following assemblies:
//
// * installing/installing-mirroring-installation-images.adoc

[id="mirror-registry-localhost{context}"]
= Mirroring on a local host with mirror registry for Red Hat Openshift

> This procedure introduces how to install the _mirror registry for Red Hat Openshift_ on a local host using the `mirror-registry` installer tool. By doing so, users can create a local host registry running on port 443 for the purpose of storing a mirror of {product-title} images.

Reviewer 3: ```suggestion
This procedure explains how to install the _mirror registry for Red Hat Openshift_ on a local host using the `mirror-registry` installer tool. By doing so, users can create a local host registry running on port 443 to store a mirror of {product-title} images.
```

[NOTE]
====

> Installing the mirror registry for Red Hat Openshift using the `mirror-registry` CLI tool makes several changes to your machine. After installation, a `/etc/quay-install` directory is created, which has installation files, local storage, and the configuration bundle. Trusted SSH keys are generated in case the deployment target is the local host, and systemd files on the host machine are set up to ensure that container runtimes are persistent. Additionally, an initial user named `init` is created with an auto-generated password. All access credentials are printed at the end of the install routine.

Reviewer 3: ```suggestion
Installing the mirror registry for Red Hat Openshift using the `mirror-registry` CLI tool makes several changes to your machine. After installation, a `/etc/quay-install` directory is created, which has installation files, local storage, and the configuration bundle. Trusted SSH keys are generated in case the deployment target is the local host, and systemd files on the host machine are set up to ensure that container runtimes are persistent. Additionally, an initial user named `init` is created with an automatically generated password. All access credentials are printed at the end of the install routine.
```

====

.Procedure

> . Download the `mirror-registry.tar.gz` package for the latest version of mirror registry found on the link:https://catalog.redhat.com/software/containers/openshift/mirror-registry-rhel8[mirror registry for Red Hat Openshift] page.

Reviewer 3: ```suggestion
. Download the `mirror-registry.tar.gz` package for the latest version of mirror registry for Red Hat OpenShift found on the link:https://catalog.redhat.com/software/containers/openshift/mirror-registry-rhel8[mirror registry for Red Hat Openshift] page.
```

Reviewer 3: Link is a 404

+
[source,terminal]
----
$ sudo ./mirror-registry install \
  --quayHostname <host_example_com> \
  --quayRoot <example_directory_name>
----

. Use the user name and password generated during installation to log into the registry by running the following command:
+
[source,terminal]
----
$ podman login -a pull-secret.txt \
  -u init \
  -p <password> \
  <host_example_com>:8443> \
  --tls-verify=false <1>
----
<1> You can avoid running `--tls-verify=false` by configuring your system to trust the generated rootCA certificates. See "Using SSL to protect connections to Red Hat Quay" and "Configuring the system to trust the certificate authority" for more information.
+
[NOTE]
====
You can also log in by accessing the UI at `\https://<host.example.com>:8443` after installation.
====

. You can mirror {product-title} images after logging in. Depending on your needs, see either the "Mirroring the {product-title} image repository" or the "Mirroring Operator catalogs for use with disconnected clusters" sections of this document.
+
[NOTE]
====

> If there are issues with images stored by mirror registry due to storage layer problems, customers are encouraged to either re-mirror the {product-title} images, or re-install mirror registry on more stable storage.

Reviewer 3: ```suggestion
If there are issues with images stored by mirror registry for Red Hat OpenShift due to storage layer problems, you can remirror the {product-title} images, or reinstall mirror registry on more stable storage.
```

// Module included in the following assemblies:
//
// * installing/installing-mirroring-installation-images.adoc

[id="mirror-registry-localhost{context}"]

> 

Reviewer 4: Remove the newline here to avoid any potential weird build errors

> = Local host mirroring with mirror registry for {product-title}

Reviewer 4: Since this is a procedural module, the title should begin with a gerund.

This procedure introduces how to install Red Hat Quay on a local host using the `mirror-registry` tool. By doing so, users can create a registry to hold a mirror of {product-title} images.

[NOTE]
====
Installing Red Hat Quay using the `mirror-registry` CLI tool makes several changes to your machine. After installation, a `/etc/quay-install` directory is created, which has installation files, local storage, and the configuration bundle. Trusted SSH keys are generated in case the deployment target is the local host, and systemd files on the host machine are set up to ensure that container runtimes are persistent. Additionally, an initial user named `init` is created with an auto-generated password. All access credentials are printed at the end of the install routine.
====

> 

Reviewer 4: I would add a `.Procedure` block title here to indicate the procedure start

+
[source,terminal]
----
$ podman login -u init \
-p <password> \
--tls-verify=false <1>
----
<1> You can avoid running `--tls-verify=false` by configuring your system to trust the generated rootCA certificates. See "Using SSL to protect connections to Red Hat Quay" and "Configuring the system to trust the certificate authority" for more information.

. You can mirror {product-title} images after logging in. Depending on your needs, see either the "Mirroring the {product-title} image repository" or the "Mirroring Operator catalogs for use with disconnected clusters" sections of this document.
+
[NOTE]
====
If there are issues with images stored by mirror registry due to storage layer problems, customers are encouraged to either re-mirror the {product-title} images, or re-install mirror registry on more stable storage.
====

. You can uninstall Red Hat Quay from your local host by running the following command:
+
[source,terminal]
----
$ sudo ./mirror-registry uninstall -v \

> --quayRoot </example/directory/name>

Reviewer 4: ```suggestion
    --quayRoot </example/directory/name>
```

// Module included in the following assemblies:
//
// * installing/installing-mirroring-installation-images.adoc

[id="mirror-registry-localhost{context}"]
= Mirroring on a local host with mirror registry for {product-title}

> This procedure introduces how to install Red Hat Quay on a local host using the `mirror-registry` tool. By doing so, users can create a registry to hold a mirror of {product-title} images.

Reviewer 2: ```suggestion
This procedure introduces how to install the mirror registry on a local host using the `mirror-registry` installer tool. By doing so, users can create a registry on the local host running on port 443 for the purpose of storing a mirror of {product-title} images.
```

Author: Slight alteration

[NOTE]
====
Installing Red Hat Quay using the `mirror-registry` CLI tool makes several changes to your machine. After installation, a `/etc/quay-install` directory is created, which has installation files, local storage, and the configuration bundle. Trusted SSH keys are generated in case the deployment target is the local host, and systemd files on the host machine are set up to ensure that container runtimes are persistent. Additionally, an initial user named `init` is created with an auto-generated password. All access credentials are printed at the end of the install routine.
====

.Procedure

. Download the `mirror-registry.tar.gz` package for the latest version of mirror registry found on the link:https://console.redhat.com/openshift/downloads[*Downloads*] page of {product-title}.

> . Install Red Hat Quay on your local host with your current user account by using the `mirror-registry` tool:

Reviewer 2: ```suggestion
. Install the mirror registry on your local host with your current user account by using the `mirror-registry` tool:
```

+
[source,terminal]
----
$ sudo ./mirror-registry install \
  --quayHostname <host_example_com> \

>   --quayRoot <example_directory_name>

Reviewer 2: These switches need explanation on what they represent.

Author: All flags have explanations in the "Flag" section of the document; it was suggested by a senior OCP writer that I don't repeat the information here when it's covered in the flag section.

----

. The Red Hat Quay console is accessible at `\https://<host.example.com>:8443` after installation. Use the user name and password generated during installation to log into the registry by running the following command:
+
[source,terminal]
----
$ podman login -u init \
  -p <password> \
  --tls-verify=false <1>
----

> <1> You can avoid running `--tls-verify=false` by configuring your system to trust the generated rootCA certificates. See "Using SSL to protect connections to Red Hat Quay" and "Configuring the system to trust the certificate authority" for more information.

Reviewer 2: Where are these mentioned doc chapters?

Author: In the 'Additional Resources' section below the Flags section. This is how OCP docs have to add additional resources; x-refs can't be added in modules because they can potentially break the build.

Author: It's also been deemed unnecessary to say something like "See xyz in the Additional Resources section of this document."

> . You can mirror {product-title} images after logging in. Depending on your needs, see either the "Mirroring the {product-title} image repository" or the "Mirroring Operator catalogs for use with disconnected clusters" sections of this document.

Reviewer 2: The login to the UI is optional. The registry is ready to use. We should directly continue with the `podman login` command then followed by link to the mirror documentation and instruct the user to proceed there with mirroring through the command line.

+
[NOTE]
====
If there are issues with images stored by mirror registry due to storage layer problems, customers are encouraged to either re-mirror the {product-title} images, or re-install mirror registry on more stable storage.
====

> . You can uninstall Red Hat Quay from your local host by running the following command:

Reviewer 2: ```suggestion
. You can uninstall the mirror registry from your local host by running the following command:
```

> . You can uninstall Red Hat Quay from your local host by running the following command:

Reviewer 2: This should not be step 5 but a separate section for uninstall. People blindly copying and pasting might accidentally follow this.

// Module included in the following assemblies:
//
// * installing/installing-mirroring-installation-images.adoc

[id="mirror-registry-localhost{context}"]

= Local host mirroring with mirror registry for {product-title}

This procedure introduces how to install Red Hat Quay on a local host using the `mirror-registry` tool. By doing so, users can create a registry to hold a mirror of {product-title} images.

[NOTE]
====
Installing Red Hat Quay using the `mirror-registry` CLI tool makes several changes to your machine. Upon installation, an `/etc/quay-install` directory is created, which contains installation files, local storage, and the config bundle. Trusted SSH keys are generated in case the deployment target is the local host, and systemd files on the host machine are set up to ensure that container runtimes are persistent. Additionally, an initial user named `init` is created with an auto-generated password. All access credentials are printed at the end of the install routine.
====

.Procedure

//placeholder link

> . Download the `mirror-registry.tar.gz` package for the latest version of mirror registry found on the link:https://github.com/quay/openshift-mirror-registry/releases[mirror registry releases page].

Reviewer 5: QE was recommended to download  `mirror-registry.tar.gz` package from a image
```
podman cp $(podman create --rm registry-proxy.engineering.redhat.com/rh-osbs/openshift-mirror-registry-rhel8:v1.0-6'):/mirror-registry.tar.gz .
$ tar -xvzf mirror-registry.tar.gz
```
But here we recommand customer to download from github release page.
Which is the correct download method?

Author: Download page is preferred method for users.

. Install Red Hat Quay on your local host with your current user account by using the `mirror-registry` tool:
+
[source,terminal]
----
$ ./mirror-registry install \
--quayHostname <host.example.com> \
--quayRoot <example/directory/name>

> ----

Reviewer 5: It is better to add `sudo` in this command line like what line 51 of current page does.
